src/cloud-resources/src/crd/materialize.rs

```diff
@@ -58,6 +58,30 @@ pub mod v1alpha1 {
         // Additional annotations and labels to include in the Certificate object.
         pub secret_template: Option<CertificateSecretTemplate>,
     }
+    #[derive(Clone, Debug, PartialEq, Deserialize, Serialize, JsonSchema)]
+    pub enum MaterializeRolloutStrategy {
+        // Default. Create a new generation of pods, leaving the old generation around until the
+        // new ones are ready to take over.
+        // This minimizes downtime, and is what almost everyone should use.
+        WaitForReady,
+
+        // WARNING!!!
+        // THIS WILL CAUSE YOUR MATERIALIZE INSTANCE TO BE UNAVAILABLE FOR SOME TIME!!!
+        // WARNING!!!
+        //
+        // Tear down the old generation of pods and promote the new generation of pods immediately,
+        // without waiting for the new generation of pods to be ready.
+        //
+        // This strategy should ONLY be used by customers with physical hardware who do not have
+        // enough hardware for the WaitForReady strategy. If you think you want this, please
+        // consult with Materialize engineering to discuss your situation.
+        ImmediatelyPromoteCausingDowntime,
+    }
+    impl Default for MaterializeRolloutStrategy {
```

Comment:
you can use `derive(Default)` and the `#[default]` variant annotation instead of deriving this by hand